thread/common/thrprof.c

```diff
@@ -1025,3 +1030,35 @@ omrthread_get_jvm_cpu_usage_info_error_recovery(void)
 		GLOBAL_UNLOCK_SIMPLE(lib);
 	}
 }
+
+intptr_t
+omrthread_get_self_thread_time(omrthread_thread_time_t *threadTime)
+{
+#if defined(LINUX)
+	struct rusage rUsage = {0};
+	int rc = getrusage(RUSAGE_THREAD, &rUsage);
+
+	if (-1 == rc) {
+		return -1;
+	}
+
+	threadTime->userTime = (SEC_TO_NANO_CONVERSION_CONSTANT * (int64_t)rUsage.ru_utime.tv_sec)
+			+ (MICRO_TO_NANO_CONVERSION_CONSTANT * (int64_t)rUsage.ru_utime.tv_usec);
+	threadTime->sysTime = (SEC_TO_NANO_CONVERSION_CONSTANT * (int64_t)rUsage.ru_stime.tv_sec)
+			+ (MICRO_TO_NANO_CONVERSION_CONSTANT * (int64_t)rUsage.ru_stime.tv_usec);
+#else /* defined(LINUX) */
+	omrthread_t self = omrthread_self();
+
+	int64_t userTime = omrthread_get_user_time(self);
+	int64_t cpuTime = omrthread_get_cpu_time(self);
```

Comment:
Given these existing API functions, what's the real value of adding this function?

If we decide to go ahead with this change, this should use `omrthread_get_self_cpu_time()` and `omrthread_get_self_user_time()` instead.

Reply:
I see what you mean. Yeah, I think the best way is to add Linux implementation to `omrthread_get_self_user_time()` so we don't need to add a new function. Then we'll be calling `omrthread_get_self_cpu_time()` and `omrthread_get_self_user_time()` on the J9 side. @tajila what do you think?

Reply:
I was hoping to avoid calling `getrusage` twice, since we need both cpu and user time. What does the support look like on other platforms, are they separate system calls or can those be bundled as well.

Reply:
Looks like on Windows and AIX they can be bundled as well.

Reply:
This is not a good idea because on some platforms `omrthread_get_self_cpu_time` get CPU time directly rather than by adding user time and system time.